### PR TITLE
Enable RemoteFlowWatcherTest

### PR DESCRIPTION
**Pull request by juhoautio**
Fixed failures. It's quick, too.

### CODE REVIEW

**Comment by juhoautio** [comment]
So.. would you like to have these tests? Looks like there's a new conflict to resolve.

**Comment by HappyRay** [comment]
Yes, this change seems to be useful.
Sorry, the team are behind in reviewing PR requests.

**Comment by juhoautio** [comment]
Rebased & fixed.

File: azkaban-exec-server/src/test/java/azkaban/execapp/event/RemoteFlowWatcherTest.java [hunk 1]
```diff
@@ -229,26 +201,25 @@ private void testPipelineLevel2(final ExecutableFlow first, final ExecutableFlow
...
 
-  private FlowRunner createFlowRunner(final File workingDir, final ExecutorLoader loader,
+  private FlowRunner createFlowRunner(final ExecutorLoader loader,
       final EventCollectorListener eventCollector, final String flowName, final int execId,
       final FlowWatcher watcher, final Integer pipeline, final Props azkabanProps)
       throws Exception {
-    final File testDir = new File("unit/executions/exectest1");
-    final ExecutableFlow exFlow =
-        prepareExecDir(workingDir, testDir, flowName, execId);
+    final File testDir = new File("../test/execution-test-data/exectest1");
```
Comment (HappyRay) [inline azkaban-exec-server/src/test/java/azkaban/execapp/event/RemoteFlowWatcherTest.java]: Please use a wrapper method.
See an example 
https://github.com/HappyRay/azkaban/blob/1cbe49cce50ffde71784e73c2f03097e83e86780/azkaban-common/src/test/java/azkaban/executor/JdbcExecutorLoaderTest.java#L973

**Comment by juhoautio** [comment]
Good that you mentioned the TemporaryFolder rule in https://github.com/azkaban/azkaban/pull/1243 – also used here now. Before this change the test was contaminating `test/execution-test-data/exectest1/` with some temporary files :)

**Comment by juhoautio** [comment]
I checked what tests failed in that build 88fb0ad. They are not related to RemoteFlowWatcherTest. 

```
azkaban.execapp.event.JobCallbackRequestMakerTest > basicPostTest FAILED
    org.junit.runners.model.TestTimedOutException: test timed out after 4000 milliseconds
        at sun.misc.Unsafe.park(Native Method)
        at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
        at java.util.concurrent.FutureTask.awaitDone(FutureTask.java:426)
        at java.util.concurrent.FutureTask.get(FutureTask.java:204)
        at azkaban.execapp.event.JobCallbackRequestMaker.makeHttpRequest(JobCallbackRequestMaker.java:153)
        at azkaban.execapp.event.JobCallbackRequestMakerTest.basicPostTest(JobCallbackRequestMakerTest.java:164)

azkaban.execapp.event.JobCallbackRequestMakerTest > unResponsiveGetTest FAILED
    org.junit.runners.model.TestTimedOutException: test timed out after 4000 milliseconds
        at sun.misc.Unsafe.park(Native Method)
        at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
        at java.util.concurrent.FutureTask.awaitDone(FutureTask.java:426)
        at java.util.concurrent.FutureTask.get(FutureTask.java:204)
        at azkaban.execapp.event.JobCallbackRequestMaker.makeHttpRequest(JobCallbackRequestMaker.java:153)
        at azkaban.execapp.event.JobCallbackRequestMakerTest.unResponsiveGetTest(JobCallbackRequestMakerTest.java:144)
```

**Comment by juhoautio** [comment]
So this should be ok.

**Comment by HappyRay** [comment]
It appears that the imports are not organized correctly. Did you follow the new contribution guide?

File: azkaban-exec-server/src/test/java/azkaban/execapp/event/RemoteFlowWatcherTest.java [hunk 2]
```diff
@@ -217,10 +203,10 @@ private void testPipelineLevel2(final ExecutableFlow first, final ExecutableFlow
         Assert.assertEquals(child.getStatus(), Status.SUCCEEDED);
         final long diff = node.getStartTime() - child.getEndTime();
         minDiff = Math.min(minDiff, diff);
```
Comment (HappyRay) [inline azkaban-exec-server/src/test/java/azkaban/execapp/event/RemoteFlowWatcherTest.java]: This logic is duplicated in the same class. It would be great if they can be consolidated. It can be a separate PR though.

**Comment by HappyRay** [comment]
It's not clear to me why the test flow uses so many jobs. Would 2 or three be sufficient?

File: azkaban-exec-server/src/test/java/azkaban/execapp/event/RemoteFlowWatcherTest.java [hunk 3]
```diff
@@ -128,26 +121,22 @@ public void testLevel1RemoteFlowWatcher() throws Exception {
...
 
-    FileUtils.deleteDirectory(workingDir1);
-    FileUtils.deleteDirectory(workingDir2);
-
     testPipelineLevel1(runner1.getExecutableFlow(), runner2.getExecutableFlow());
   }
 
-  @Ignore
   @Test
   public void testLevel2DiffRemoteFlowWatcher() throws Exception {
```
Comment (HappyRay) [inline azkaban-exec-server/src/test/java/azkaban/execapp/event/RemoteFlowWatcherTest.java]: What's the difference between this test and the previous one testLevel1RemoteFlowWatcher? The only difference I can see is that the test flow file has two additional jobs. 
Could you add a comment to describe the purpose of this test or remove it if it is not necessary?

**Comment by HappyRay** [comment]
It would be great if you can fix the code duplication for prepareExecDir and createFlowRunner too. They can be in a separate PR though. Intellij should be able to warn and help with this refactoring.

**Comment by juhoautio** [comment]
Applied save actions

File: azkaban-exec-server/src/test/java/azkaban/execapp/event/RemoteFlowWatcherTest.java (hunk 3 above)
Comment (juhoautio) [reply azkaban-exec-server/src/test/java/azkaban/execapp/event/RemoteFlowWatcherTest.java]: I haven't analyzed it to see what the impact is, but there's a difference between using one of:
- [exec1-mod.flow](https://github.com/azkaban/azkaban/blob/master/test/execution-test-data/exectest1/exec1-mod.flow)
- [exec1.flow](https://github.com/azkaban/azkaban/blob/master/test/execution-test-data/exectest1/exec1.flow)

Here's the diff:

```
diff test/execution-test-data/exectest1/exec1-mod.flow test/execution-test-data/exectest1/exec1.flow
33,35d32
<   },{
<     "source" : "job7",
<     "target" : "job9"
40,43d36
<   },
<   {
<     "source" : "job9",
<     "target" : "job10"
129,138d121
<     "id" : "job9",
<     "jobType" : "java",
<     "layout" : {
<       "level" : 0
<     },
<     "jobSource" : "job9.job",
<     "expectedRuntime" : 1
<   },
<   {
<     "propSource" : "prop2.properties",
```

I would assume it has some purpose.. I don't see why it should be remove without knowing if it really is redundant. But to me that's outside the scope of this PR. I'm just making the existing tests run.

**Comment by juhoautio** [comment]
As far as I can see, @HappyRay's open comments are about refactoring that should be done separately. Ready to merge?

**Comment by HappyRay** [comment]
Thanks!

**Comment by HappyRay** [comment]
It would be great if you can follow up with some additional improvements we discussed here.
